### PR TITLE
adds `tmpdir` option to fix #203

### LINKED ISSUE #203: Add configurable option for the temp build directory
Using this command line:

    $ electron-packager . UF --platform=win32 --overwrite --arch=ia32 --version=0.34.2

My app builds just fine if I executed on a subset of it (e.g. 2 of 41 chapters of an educational resource).

However, a full build fails silently (20G of product, most of that videos).

Nervous about shipping product in 2 weeks... so close to working!

How do I debug this?

### PR DESCRIPTION
**Pull request by eugirdor**
Custom temporary directory can be supplied in `opts.tmpdir` or from the command line switch `--tmpdir`.  This address issue #203.

### CODE REVIEW

**Comment by malept** [comment]
Thanks for the PR! Perhaps add a test to make sure the option works as expected?

**Comment by eugirdor** [comment]
@malept Any idea what might have caused the one test to fail?  Doesn't seem related to my changes.

**Comment by malept** [comment]
That particular test intermittently fails (and I don't know why). Don't worry about it.

**Comment by eugirdor** [comment]
Alright, thanks.  I'll add a test a bit and push again.

**Comment by eugirdor** [comment]
@malept Test added

File: readme.md [hunk 1]
```diff
@@ -207,6 +207,10 @@ If the file extension is omitted, it is auto-completed to the correct extension
 
   Whether SSL certificates are required to be valid when downloading Electron. **Defaults to `true`**.
 
+  `tmpdir` - *String*
```
Comment (malept) [inline readme.md]: Minor nit - this is the wrong indentation. There should be no spaces between the beginning of the line and the option name. The description's indentation is fine.

**Comment by malept** [comment]
Would you mind rebasing and squashing the commits after you fix the minor docs issue?

**Comment by eugirdor** [comment]
@malept Whitespace fixed and commits squashed

**Comment by malept** [comment]
Thanks!

**Pull request merged**